From 1f6a663c88b6cd2126946430cebfc42e8daaca0d Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@cloudera.com>
Date: Thu, 11 Aug 2016 17:20:16 -0700
Subject: [PATCH 1729/2748] HDFS-10347. Namenode report bad block method
 doesn't log the bad block or datanode.
 Contributed by Rushabh Shah.

(cherry picked from commit 7da540d03eccc2f97950bf47e8b35ce8c889d1e0)
Updated CHANGES.txt

(cherry picked from commit 6347f8f4d455361421e526ff899654cdc4287a5b)

Change-Id: Iae998d63dbba484b4b4160c6351217af0b829a85
---
 .../hadoop/hdfs/server/namenode/FSNamesystem.java  |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java
index aaed0cf..08b168f 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java
@@ -7113,7 +7113,6 @@ private INodeFile checkUCBlock(ExtendedBlock block,
    */
   void reportBadBlocks(LocatedBlock[] blocks) throws IOException {
     checkOperation(OperationCategory.WRITE);
-    NameNode.stateChangeLog.info("*DIR* reportBadBlocks");
     writeLock();
     try {
       checkOperation(OperationCategory.WRITE);
@@ -7122,6 +7121,8 @@ void reportBadBlocks(LocatedBlock[] blocks) throws IOException {
         DatanodeInfo[] nodes = blocks[i].getLocations();
         String[] storageIDs = blocks[i].getStorageIDs();
         for (int j = 0; j < nodes.length; j++) {
+          NameNode.stateChangeLog.info("*DIR* reportBadBlocks for block: {} on"
+              + " datanode: {}", blk, nodes[j].getXferAddr());
           blockManager.findAndMarkBlockAsCorrupt(blk, nodes[j],
               storageIDs == null ? null: storageIDs[j], 
               "client machine reported it");
-- 
1.7.9.5

